commands/acr-build.ts

```diff
@@ -0,0 +1,99 @@
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { QuickBuildRequest } from "azure-arm-containerregistry/lib/models";
+import { Registry } from 'azure-arm-containerregistry/lib/models';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { BlobService, createBlobServiceWithSas } from "azure-storage";
+import * as vscode from "vscode";
+import { ImageNode } from "../explorer/models/imageNode";
+import { AzureUtilityManager } from "../utils/azureUtilityManager";
+import { acquireResourceGroup, acquireSubscription, quickPickACRRegistry } from './utils/quick-pick-azure';
+let tar = require('tar');
+let fs = require('fs');
+let os = require('os');
+let url = require('url');
+
+export async function queueBuild(context?: ImageNode): Promise<void> {
+
+    console.log("Obtaining Subscription and Client");
+    let subscription = await acquireSubscription();
+    let client = AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
+    const resourceGroupClient = new ResourceManagementClient(AzureUtilityManager.getInstance().getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+
+    let resourceGroup = await acquireResourceGroup(subscription, resourceGroupClient);
+    let resourceGroupName = resourceGroup.name;
+
+    let registry: Registry = await quickPickACRRegistry(subscription, resourceGroupName);
+    let registryName = registry.name;
+
+    console.log("Setting up temp file with 'sourceArchive.tar.gz' ");
+    let tarFilePath = url.resolve(os.tmpdir(), 'sourceArchive.tar.gz');
```

Comment:
The filename is fixed. You need to make sure
1) A new build will overwrite the old tarfile
2) It's better to delete the tarfile after upload.
2) Ideally you should use a random tarfile name in each build to avoid confliction. For example, if I want to queue the builds using different Dockerfile in the project. at the same time